Sources/JavaScriptKit/FundamentalObjects/JSClosure.swift

```diff
@@ -40,26 +73,27 @@ public class JSClosure: JSObject, JSClosureProtocol {
     }
 
     public init(_ body: @escaping ([JSValue]) -> JSValue) {
-        self.hostFuncRef = closureRef
-        closureRef += 1
-
-        // Retain the given body in static storage by `closureRef`.
-        sharedClosures[self.hostFuncRef] = body
-
-        // Create a new JavaScript function which calls the given Swift function.
+        // 1. Fill `id` as zero at first to access `self` to get `ObjectIdentifier`.
+        super.init(id: 0)
+        let objectId = ObjectIdentifier(self)
```

Comment:
I switched away from using `ObjectIdentifier` because:

```swift
foo(JSClosure { _ in "this closure" })
bar(JSClosure { _ in "and this closure" })
```

…ended up having the same `ObjectIdentifier` value. The monotonically increasing global ID works around that.

Reply:
I think that behavior seems natural.  In what use cases is that behavior useful?

Reply:
Adding multiple event listeners to a DOM node, for example.

Reply:
In that case, the closure instances are retained by host environment, so the Swift instances are also retained and they should have different identifies at a time

Reply:
The closure itself (`{ _ in ... }`) is retained by the host environment, but the `JSClosure` wrapper is immediately deallocated because it is not retained by anything.

Reply:
Ah, I see. So we should retain the JSClosure instance itself.